File: src/main/java/sh/twenty/octane/commands/OctaneCommand.java
package sh.twenty.octane.commands;

import org.bukkit.entity.Player;
import sh.twenty.octane.Main;

public class OctaneCommand extends PlayerCommand {
    public OctaneCommand(Main plugin){
        super(plugin, "octane.admin");
    }
    @Override
    public boolean execute(Player player, String[] args){
        if (args.length == 1 && args[0].equalsIgnoreCase("reload")){
            plugin.getConfigManager().reloadConfigs();

            sendPrefixedMessage(player, "&7The configuration has been reloaded.");
            return true;
        }

        sendPrefixedMessage(player, "&7Usage: /octane reload");
        return true;
    }
}

File: src/main/kotlin/sh/twenty/octane/commands/OctaneCommand.kt
package sh.twenty.octane.commands

import dev.jorel.commandapi.kotlindsl.commandAPICommand
import dev.jorel.commandapi.kotlindsl.literalArgument
import dev.jorel.commandapi.kotlindsl.playerExecutor
import sh.twenty.octane.OctanePlugin

class OctaneCommand(private val plugin: OctanePlugin) {
    fun register() = commandAPICommand("octane") {
        withPermission("octane.admin")
        literalArgument("reload")
        playerExecutor { player, _ ->
            plugin.configManager.reload()
            val msgs = plugin.configManager.messages
            player.sendRichMessage(msgs.prefix + msgs.configReloaded)
        }
    }
}
